### PR TITLE
allow unbounded ram tasks windows

### PR DESCRIPTION
**Pull request by adnxn**
### Summary

carrying over https://github.com/aws/amazon-ecs-agent/pull/2142 and testing. see https://github.com/aws/amazon-ecs-agent/issues/1144 and https://github.com/aws/amazon-ecs-agent/pull/2142 for context.

### Testing

New tests cover the changes: 

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -170,6 +170,7 @@ additional details on each available environment variable.
 | `ECS_CGROUP_PATH` | `/sys/fs/cgroup` | The root cgroup path that is expected by the ECS agent. This is the path that accessible from the agent mount. | `/sys/fs/cgroup` | Not applicable |
 | `ECS_CGROUP_CPU_PERIOD` | `10ms` | CGroups CPU period for task level limits. This value should be between 8ms to 100ms | `100ms` | Not applicable |
 | `ECS_ENABLE_CPU_UNBOUNDED_WINDOWS_WORKAROUND` | `true` | When `true`, ECS will allow CPU unbounded(CPU=`0`) tasks to run along with CPU bounded tasks in Windows. | Not applicable | `false` |
+| `ECS_ENABLE_RAM_UNBOUNDED_WINDOWS_WORKAROUND` | `true` | When `true`, ECS will allow tasks with no memory reservation to run along with memory bounded tasks in Windows. To run a memory unbounded task, omit the memory hard limit and set any memory reservation, it will be ignored. | Not applicable | `false` |
```
Comment (sharanyad) [inline README.md]: what would be a good counterpoint for calling this `ECS_ENABLE_MEMORY_UNBOUNDED_WINDOWS_WORKAROUND` ?

File: agent/api/task/task_windows_test.go [hunk 2]
```diff
@@ -254,6 +257,52 @@ func TestCPUPercentBasedOnUnboundedEnabled(t *testing.T) {
...
+			},
+		},
+		PlatformFields: PlatformFields{
+			RamUnbounded: false,
+		},
+	}
+
+	// With RamUnbounded set to false, MemoryReservation is not overridden
+	config, configErr := testTask.DockerHostConfig(testTask.Containers[0], dockerMap(testTask), defaultDockerClientAPIVersion)
+	if configErr != nil {
```
Comment (sharanyad) [inline agent/api/task/task_windows_test.go]: this can be assert. same below

File: agent/config/config_windows.go [hunk 3]
```diff
@@ -118,8 +119,10 @@ func (cfg *Config) platformOverrides() {
 	cfg.TaskCPUMemLimit = ExplicitlyDisabled
 
 	cpuUnbounded := utils.ParseBool(os.Getenv("ECS_ENABLE_CPU_UNBOUNDED_WINDOWS_WORKAROUND"), false)
+	ramUnbounded := utils.ParseBool(os.Getenv("ECS_ENABLE_RAM_UNBOUNDED_WINDOWS_WORKAROUND"), false)
```
Comment (sharanyad) [inline agent/config/config_windows.go]: I believe these need unit tests?

File: README.md (hunk 1 above)
Comment (petderek) [reply README.md]: s/allow tasks with no memory reservation/ignore the memory reservation parameter (soft limit)/

Comment (adnxn) [reply README.md]: im okay with this

File: agent/api/task/task_windows_test.go (hunk 2 above)
Comment (petderek) [reply agent/api/task/task_windows_test.go]: you can use require.NoError(t, err) if you want the exit behavior you get from t.fatal

File: agent/config/config_windows.go (hunk 3 above)
Comment (adnxn) [reply agent/config/config_windows.go]: added tests to check default config

File: agent/api/task/task_windows_test.go (hunk 2 above)
Comment (adnxn) [reply agent/api/task/task_windows_test.go]: https://github.com/stretchr/testify/issues/562 

😞

**Pull request merged**